vue/components/molecules/restrictions-alert/restrictions-alert.vue

```diff
@@ -77,14 +77,20 @@ export const RestrictionsAlert = {
             visible: false
         };
     },
-    mounted: function() {
-        this.$bus.bind("restrictions", (changes, newPart) => {
+    created: function() {
+        this.onRestrictions = this.$bus.bind("restrictions", (changes, newPart) => {
             this.visible = changes.length > 0;
         });
+
+        this.onUndo = this.$bus.bind("undo", () => this.close());
+    },
+    destroyed: function() {
+        if (this.onRestrictions) this.$bus.unbind("restrictions", this.onRestrictions);
+        if (this.onUndo) this.$bus.unbind("undo", this.onUndo);
     },
     methods: {
         undo() {
-            this.visible = false;
+            this.close();

```

Comment:
Is this needed?

Reply:
Not really needed but there is an Incoherence on how the aler is closed.
In [line 9](https://github.com/ripe-tech/ripe-commons-pluginus/blob/f1138832f1e66c17b538b97ab9650cdce844aa87/vue/components/molecules/restrictions-alert/restrictions-alert.vue#L9) the method `close()` is called and in line 87, the alert is closed with `this.visible = false;` instead of using the `close() `method.

For reference, the `close()` method: https://github.com/ripe-tech/ripe-commons-pluginus/blob/6dcff15a53b757b5e126a88ed099c1c54ccca6e4/vue/components/molecules/restrictions-alert/restrictions-alert.vue#L90

Reply:
What I meant was: you don't need neither `this.visible = false` nor `this.close()` because `undo` will now do that.

Reply:
Oh, I see what you mean now, you are asking if the `this.$bus.trigger("undo");` is going to close the alert, making this line not needed.

Yes you are right 👍